### ISSUE TITLE
Reload config doesn't work for Custom Token/Appenders

### ISSUE THREAD

**Issue by roshansalvi** [nomiddlename/log4js-node#497]
Is there any way where I can reload the log4js config by passing object? Currently if I pass object it gives me warning "'Ignoring configuration reload parameter for "object" configuration'" and doesn't reload the configuration.  I am passing object because I have custom token that I am adding programmatically.  Also I need to reload the configuration so that I dont have to stop and start server if I change the log level.

It will be great help if you can help me to resolve this issue. ?  Thanks in advance.

**Comment by nomiddlename** [comment]
If you're not using a file for config, how will log4js know that the config has changed? How do you change the log level?

**Comment by roshansalvi** [comment]
Okay Let me rephrase the question.   I have a below json file : 
`
  "appenders": [
    {
      "type": "file",
      "filename": "mc.log",
      "maxLogSize": 10485760,
      "backups": 10,
      "layout": {
        "type": "pattern",
        "pattern": "%d{ISO8601_WITH_TZ_OFFSET} [%x{requestContext}] [%z] %p %c - %x{message}%n"
      }
    }
    }
  ],
  "levels": {
    "horizon": "DEBUG"
  }
}`

While initializing log4js.  I read this json file (basically parse it ) and then add programatically tokens for 'requestContext' and 'message'.  This way I have an object now which I am passing as 

`log4js.configure(log4jsConfig, {cwd: logFileFolder, reloadSecs: 10});`

Now, I would like to know is it possible to reload the config in above way.   If not how do I pass the custom token 'requestContext' and 'message' in json file so that I can reload the config.

I hope I am able to explain the problem I am facing.

**Comment by nomiddlename** [comment]
Ok, I understand now. I would suggest looking at using a library like [watchr](https://www.npmjs.com/package/watchr) to monitor your JSON file and reconfigure log4js (call `log4js.shutdown` and then `log4js.configure`) when it changes. In fact, I have removed configuration reloading from version 2.x of log4js because it causes too many problems (resources not closed properly, etc) - better to use watchr or a similar library.

**Comment by roshansalvi** [comment]
Thanks a lot for your suggestion.  Much appreciated.
